WordPress/gutenberg
# Post Comments Form: Add button that enables commenting to warning

## What?
Add a button to the Post Comments Form block that allows the user to enable commenting when it's disabled.
Part of #40614; also the first experiment to explore whether a solution like this makes sense.

## Why?
Remove frustration from the user coming from a warning message that doesn't give them agency to remediate the problem. See #40614 for more context.

## How?
In the Post Comments Form block warning that states that commenting is not allowed, add a button that allows the user to enable it.

## Testing Instructions
- Create a new post.
- Open the inspector sidebar for the Post.
- Disable comments for the post. 
![image](https://user-images.githubusercontent.com/96308/172672995-29733e62-d42f-4429-b511-163d8508f007.png)
- Add the Comments block to the post.
- Verify that the warning that states that comments aren't allowed includes a button to enable them.
- Click that button and verify that the warning disappears, and the comment form appears.

_Note that this might not be the most interesting case from the list in #40614, since the "Allow comments" option is otherwise fairly easily accessible from the Post sidebar. However, we might apply the same principle (button in warning) to other cases where the corresponding setting isn't as readily availably (but e.g. somewhere in wp-admin > Settings > Discussion, see e.g. #41308)._

## Screenshots or screencast

![enable-comments](https://user-images.githubusercontent.com/96308/172669858-2e223b3a-988c-4108-8148-fe0fa85b5932.gif)

## Open questions

- Do we need to limit this based on user capabilities? (It seems like the [option in the Post sidebar](https://github.com/WordPress/gutenberg/blob/5e09117a2bab1f08e529aa9c0111f31efe0aca6f/packages/editor/src/components/post-comments/index.js) isn't really restricted based on caps, so maybe they're available to anyone with editing capabilities.)
- Separations of concerns, see https://github.com/WordPress/gutenberg/pull/41603#discussion_r892728680.
